packages/react-admin/src/mui/field/ReferenceField.js

```diff
@@ -86,36 +87,23 @@ export class ReferenceField extends Component {
         if (!referenceRecord && !allowEmpty) {
             return <LinearProgress />;
         }
-        const rootPath = basePath
-            .split('/')
-            .slice(0, -1)
-            .join('/');
+        const rootPath = basePath.replace(`/${resource}`, '');
```

Comment:
Smart! But what if the resource name is `edit` (as in video edit) or `show` (as in trade show)? You should handle this special case

Reply:
I checked that and when in an edit view, basepath is `/${resource}`, not `/${resource}/edit`

Reply:
Added tests for those cases